File: src/dataset/cpp/triangle_exercism-stungeye/openAI/ChatGPT4_triangle_v3.cpp
#include "triangle.h"
#include <algorithm>

namespace triangle {
  flavor kind(double a, double b, double c) {
    if (a <= 0 || b <= 0 || c <= 0) {
      throw std::domain_error("All triangle sides must have a length greater than zero.");
    }

    double sides[] = {a, b, c};
    std::sort(sides, sides + 3);

    if (sides[0] + sides[1] < sides[2]) {
      throw std::domain_error("Combined length of any two triangle sides must be greater or equal to length of third side.");
    }

    if (sides[0] == sides[2]) {
      return flavor::equilateral;
    } else if (sides[0] == sides[1] || sides[1] == sides[2]) {
      return flavor::isosceles;
    }

    return flavor::scalene;
  }
}  // namespace triangle

File: src/dataset/typescript/triangle_exercism-wackerow/openAI/ChatGPT4_triangle_v3.ts
export default class Triangle {
  private _sides: number[];

  constructor(...sides: number[]) {
    this._sides = sides;
    if (
      sides.length !== 3 ||
      sides.some(side => side <= 0) ||
      sides[0] + sides[1] <= sides[2] ||
      sides[1] + sides[2] <= sides[0] ||
      sides[2] + sides[0] <= sides[1]
    ) {
      throw 'Impossible triangle';
    }
  }

  public kind(): string {
    const [a, b, c] = this._sides;
    if (a === b && b === c) return 'equilateral';
    if (a === b || b === c || c === a) return 'isosceles';
    return 'scalene';
  }
}